js_modules/dagit/packages/core/src/asset-graph/useAssetGraphData.tsx

```diff
@@ -130,6 +133,34 @@ const removeEdgesToHiddenAssets = (graphData: GraphData) => {
   }
 };
 
+export const calculateGraphDistances = (items: AssetGraphQueryItem[], assetKey: AssetKey) => {
```

Comment:
Would you mind writing a test for this? Also, is there any possibility that it could have some super pathological outcome on extremely wide/deep graphs?

Reply:
Yep will do 🙌